### ISSUE TITLE
关于paddlepaddle2.3.2与之后版本(2.4.0开始)不兼容的问题

### ISSUE THREAD

**Issue by Capiday** [PaddlePaddle/Paddle#62632]
### bug描述 Describe the Bug

我搭建了一个简易的CNN模型，使用的是版本是2.3.2，我想把库升级到2.4.0以上使用。
但是相同的代码在2.4.0 - 2.6.0上跑出来的结果(图2)与2.3.2跑出来的结果(图1)完全不一样。
我将2.6.0环境下训练出的模型进行保存，在2.3.2的环境中跑出来的结果与2.3.2的一致，与2.6.0完全不一样。
![1710150285550](https://github.com/PaddlePaddle/Paddle/assets/58663152/2fe09fcc-76c7-4c77-9373-34488198af00)
![1710150327273](https://github.com/PaddlePaddle/Paddle/assets/58663152/8d360584-f4a7-4071-a477-3d8bf8619213)

### 其他补充信息 Additional Supplementary Information

以下为我定义的模型与训练过程，预测过程
我测试的版本从2.2.2到2.6.0
2.2.2、2.3.0、2.3.1与2.3.2的结果一致，与之后的版本完全不一样
2.4.0 - 2.6.0的结果一致，与2.3.2以及之前测试的版本完全不一样

'''定义模型'''
class CNN(paddle.nn.Layer):
    def __init__(self):
        super(CNN, self).__init__()
        self.conv1 = paddle.nn.Conv2D(in_channels=11, out_channels=21, kernel_size=3, stride=1, padding=1, padding_mode='reflect')
        self.conv2 = paddle.nn.Conv2D(in_channels=21, out_channels=12, kernel_size=3, stride=1, padding=1, padding_mode='reflect')
        self.conv3 = paddle.nn.Conv2D(in_channels=12, out_channels=6,  kernel_size=3, stride=1, padding=1, padding_mode='reflect')

    def forward(self, x):
        x = self.conv1(x)
        x = paddle.nn.functional.relu(x)
        x = self.conv2(x)
        x = paddle.nn.functional.relu(x)
        x = self.conv3(x)
        x = paddle.nn.functional.sigmoid(x)
        return x

'''定义超参数'''
learning_rate = 0.00086
epochs = 2

'''创建模型、优化器和损失函数'''
model = CNN()
optimizer = paddle.optimizer.Adam(learning_rate=learning_rate, parameters=model.parameters())
mse_loss = paddle.nn.MSELoss()

'''定义训练和测试函数'''
def train(model, optimizer, train_loader):
    model.train()
    train_loss = 0

    for batch_id, data in enumerate(train_loader()):
        x_data, y_data = data
        predicts = model(x_data)
        loss = mse_loss(predicts, y_data)
        loss.backward()
        optimizer.step()
        optimizer.clear_grad()
        train_loss += loss.item()

    return train_loss / len(train_loader)

def test(model, test_loader):
    model.eval()
    test_loss = 0

    with paddle.no_grad():
        for batch_id, data in enumerate(test_loader()):
            x_data, y_data = data
            predicts = model(x_data)
            loss = mse_loss(predicts, y_data)
            test_loss += loss.item()

    return test_loss / len(test_loader)

print('模型训练中:')
for epoch in range(epochs):
    train_loss = train(model, optimizer, train_loader)
    test_loss = test(model, test_loader)
    print("Epoch [{}/{}], Train Loss: {:.5f}, Test Loss: {:.5f}".format(epoch + 1, epochs, train_loss, test_loss))
paddle.save(model.state_dict(), 'model_params_232.pdparams')

**Comment by JZ-LIANG** [comment]
2.x 的大版本升级，是有可能改动 kernel 实现的，所以不能保证相同模型的计算结果(比如 loss 数值) 完全相同，但数值在有效位应该近似(allclose)。
但是看你给图片来看，感觉下图是模型加载上就出现了问题，建议先在 2.3 和 2.6 下打印一下模型参数，查看数值是否相等。

如果不想等，那么 2.6 框架load  2.3 checkpoint需要用特殊的适配。比如将： 2.3 的checkpoint 先转成 numpy， 在通过 numpy load 到 2.6 的模型中。
可以参考：https://www.paddlepaddle.org.cn/documentation/docs/zh/faq/save_cn.html

**Comment by Capiday** [comment]
@JZ-LIANG 
问题已经解决，是内存问题引起的

**Comment by Capiday** [comment]
<img width="504" alt="1720013130058" src="https://github.com/PaddlePaddle/Paddle/assets/58663152/275e7cb5-b4b6-4130-8b82-54a83e4530be">
<img width="455" alt="1720013841420" src="https://github.com/PaddlePaddle/Paddle/assets/58663152/ba13f722-5a03-4b15-8971-2832d87987c6">
<img width="448" alt="1720015270267" src="https://github.com/PaddlePaddle/Paddle/assets/58663152/8ee25fd2-fe3b-40e2-a551-8874e96ce67f">

**Comment by Capiday** [comment]
最上面的图是用2.3.2版本的paddlepaddle对数据进行处理的结果，X_1是需要被模型处理的数据，我使用numpy.save将其保存为.npy文件后有4.2个G
中间的图是用2.4.2版本的paddlepaddle对数据进行处理的结果，可以看见结果已经出了问题，所有值都在0.488-0.492之间
最下面的图是将X_1拆分后使用2.4.2版本的paddlepaddle对数据进行处理的结果，对比上面两张图已经恢复正常

**Comment by Capiday** [comment]
我还将拆分后的数据跑了2.4.0和2.4.1，结果与2.3.2保持一致，这里不放图了
对此，我不是很理解
1）对比2.3.2甚至更早的2.2.2，2.4.0开始为什么要对模型可以处理的数据量大小进行限制，这一改动也没有写进开发日志里
2）如果模型处理不了这么大的数据量，为什么没用warning或者error提醒

希望有关开发者可以注意到以上问题并修改，也衷心的祝愿paddlepaddle可以越做越好，~爱来自中国澳门
